tests/console/yast2_samba.pm

```diff
@@ -14,258 +14,338 @@ use strict;
     }
 
-    zypper_call("in yast2-auth-server yast2-ldap openldap2 openldap2-client krb5-server krb5-client tdb-tools", timeout => 240);
-
-    script_run("yast2 auth-server; echo yast2-auth-server-status-\$? > /dev/$serialdev ", 0);
-    # check ldap server configuration started
-    assert_screen([qw(yast2_ldap_configuration_startup yast2_still_susefirewall2)], 60);
-    if (match_has_tag 'yast2_still_susefirewall2') {
-        record_soft_failure 'bsc#1064405';
-        send_key 'alt-c';
+    if ($error ne "") {
+        assert_script_run("echo \"$error\" > /tmp/failed_smb_directives.log");
+        die 'Missing smb.conf directives';
     }
-    # only older version like SLES 12, Leap 42.3 as well as TW should still check the needle
-    if (is_sle('<15') || is_leap('<15.0')) {
-        assert_screen 'yast2_ldap_configuration_general-setting_firewall', 60;
+}
+
+sub setup_yast2_ldap_server {
+    my %ldap_options_to_dirs = (
+        f => 'fqdn',
+        i => 'dir_instance',
+        t => 'dir_suffix',
+        d => 'dir_manager_dn',
+        r => 'dir_manager_passwd',
+        e => 'dir_manager_passwd',
+        s => 'ca_cert_pem',
+        v => 'srv_cert_key_pkcs12'
+    );
+
+    # setup FQDN hostname
+    assert_script_run("hostname openqa.ldap.test");
+    assert_script_run("echo '127.0.0.1 openqa.ldap.test openqa.ldap' > /etc/hosts");
+    # create CA cert & pkcs12 cert for LDAP
+    assert_script_run("openssl req -newkey rsa:2048 -nodes -keyout /etc/key.pem -x509 -days 365 \ "
+          . "-out /etc/certificate.pem -subj '/C=DE/ST=Bayern/L=Nuremberg/O=Suse/OU=QA/CN=localhost/emailAddress=admin\@localhost'");
+    assert_script_run("openssl pkcs12 -inkey /etc/key.pem -in /etc/certificate.pem -export -out /etc/certificate.p12 -passout pass:");
+
+    script_run("yast2 ldap-server; echo yast2-ldap-server-status-\$? > /dev/$serialdev ", 0);
+
+    # setup LDAP
+    wait_still_screen(2);
+    foreach (sort keys %ldap_options_to_dirs) {
+        wait_screen_change { send_key "alt-$_" };
+        type_string($ldap_directives{$ldap_options_to_dirs{$_}} . "\n");
     }
-    send_key 'alt-e';
 
-    # configure stand-alone ldap server
-    assert_screen 'yast2_ldap_configuration_stand-alone';
-    send_key 'alt-n';
+    wait_screen_change { send_key "alt-o" };
+    wait_screen_change { send_key "alt-y" };
+}
 
-    assert_screen 'yast2_ldap_configuration_stand-alone_tls';
-    # move to next page basic database settings and set base dn, ldap admin password
-    wait_screen_change { send_key 'alt-n' };
-    wait_screen_change { send_key 'alt-s' };
-    type_string 'dc=qa, dc=suse, dc=de';
+sub setup_ldap_in_samba {
+    ## swith to LDAP Settings
+    send_key 'alt-l';
+    assert_screen 'yast2_samba-server_ldap-settings';
+    send_key 'alt-b';
+    assert_screen 'yast2_samba-server_ldap_value_rewritten';
+    wait_screen_change { send_key 'alt-y' };
+    wait_screen_change { send_key 'alt-e' };
+    type_string 'ldap://localhost:389';
```

Comment:
Similar case here, I believe it's better to introduce couple of needles to sync key presses and assert that they reached SUT